tests/attested_tls/oe_enclave/enc/tlssrv.cpp

```diff
@@ -0,0 +1,826 @@
+             strlen(_pers))) != 0)
+    {
+        _put_mbedtls_err(err, r, "mbedtls_entropy_func()");
+        ret = r;
+        goto done;
+    }
+
+    _started = true;
+    ret = 0;
+
+done:
+
+    if (ret != 0)
+    {
+        mbedtls_entropy_init(&_entropy);
+        mbedtls_ctr_drbg_init(&_ctr_drbg);
+    }
+
+    return ret;
+}
+
+int tlssrv_shutdown(tlssrv_err_t* err)
+{
+    int ret = -1;
+
+    _clear_err(err);
+
+    if (!_started)
+    {
+        _put_err(err, "not started");
+        goto done;
+    }
+
+    mbedtls_entropy_free(&_entropy);
+    mbedtls_ctr_drbg_free(&_ctr_drbg);
+
+done:
+
+    return ret;
+}
+
+static int _get_cert_and_private_key(
+    mbedtls_x509_crt* crt,
+    mbedtls_pk_context* pk,
+    tlssrv_err_t* err)
+{
+    int ret = -1;
+    uint8_t* cert_data = NULL;
+    size_t cert_size;
+    uint8_t* private_key_data = NULL;
+    size_t private_key_size;
+    const char COMMON_NAME[] = "CN=Open Enclave SDK,O=OESDK TLS,C=US";
+
+    _clear_err(err);
+
+    if (!crt || !pk)
+        goto done;
+
+    if (generate_attested_credentials(
+            COMMON_NAME,
+	        0,
+            &cert_data,
+            &cert_size,
+            &private_key_data,
+            &private_key_size) != 0)
+    {
+        goto done;
+    }
+
+    /* Convert the certificate from DER to internal format */
+    if (mbedtls_x509_crt_parse_der(crt, cert_data, cert_size) != 0)
+    {
+        goto done;
+    }
+
+    /* Convert the private key from PEM to internal format */
+    if (mbedtls_pk_parse_key(pk, private_key_data, private_key_size, NULL, 0) !=
+        0)
+    {
+        goto done;
+    }
+
+    ret = 0;
+
+done:
+    return ret;
+}
+
+/* The mbedtls debug tracing function */
+static void _mbedtls_dbg(
+    void* ctx,
+    int level,
+    const char* file,
+    int line,
+    const char* str)
+{
+    (void)level;
+    (void)ctx;
+
+    printf("_mbedtls_dbg.server: %s:%04d: %s", file, line, str);
+}
+
+static oe_result_t _enclave_identity_verifier(
+    oe_identity_t* identity,
+    void* arg)
+{
+    tlssrv_t* srv = (tlssrv_t*)arg;
+
+    if (!identity || !srv || !srv->verify_identity)
+        return OE_VERIFY_FAILED;
+
+    return srv->verify_identity(
+        srv->verify_identity_arg,
+        identity->unique_id,
+        OE_UNIQUE_ID_SIZE,
+        identity->signer_id,
+        OE_SIGNER_ID_SIZE,
+        identity->product_id,
+        OE_PRODUCT_ID_SIZE,
+        identity->security_version);
+}
+
+static int _cert_verify_callback(
+    void* data,
+    mbedtls_x509_crt* crt,
+    int depth,
+    uint32_t* flags)
+{
+    int ret = MBEDTLS_ERR_X509_CERT_VERIFY_FAILED;
+    unsigned char* cert_buf = NULL;
+    size_t cert_size = 0;
+
+    (void)data;
+    (void)depth;
+
+    *flags = (uint32_t)MBEDTLS_ERR_X509_CERT_VERIFY_FAILED;
+
+    cert_buf = crt->raw.p;
+    cert_size = crt->raw.len;
+
+    if (cert_size <= 0)
+        goto done;
+
+    oe_result_t r;
+    if ((r = oe_verify_attestation_certificate(
```

Comment:
If you use type 1 certificate from sgx-lkl side, then you need the new API introduced in my recent PR: [`oe_verify_attestation_certificate_with_evidence`](https://github.com/openenclave/openenclave/blob/37b2a1d0a41d93ca5c816f17ba23130c6e2f7831/tests/attestation_plugin_cert/README.md).

Alternatively, you could use type 0 certificate on sgx-lkl side, and continue using `oe_verify_attestation_certificate`.

Either way is fine with me. But remember our ultimate goal is to generate a type 2 certificate on SGX-LKL side, which contains EEID evidence. When the functionality is implemented in SGX-LKL, we have to switch to `oe_verify_attestation_certificate_with_evidence` here.

Reply:
I wonder how could your test case passed the pipeline with this obvious mismatch between the certificate and the verifier. Could you double check the log to see if this test case is indeed run?

Reply:
@jxyang this test is marked for nightly run. So this test doesn't run when PR submitted.

Reply:
@jxyang is that PR the same branch to generate the cert?

Reply:
I switch back to type 0 cert because the current oe branch not support yet. I will update it once sgxlkl-support branch have all of the changes.